slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala

```diff
@@ -172,6 +186,175 @@ trait JdbcTypesComponent extends RelationalTypesComponent { self: JdbcProfile =>
+        value match {
+          case null => "NULL"
+          case _ =>  "'" + value.toString + "'"
+        }
+      }
+    }
+
+    class LocalDateTimeJdbcType extends DriverJdbcType[LocalDateTime] {
+      /**
+       * Stores the [[LocalDateTime]] as a 'VARCHAR' on databases, like SQLite' with
+       * no specific 'TIME' implementations. An example persisted value would be
+       * '2015-10-29T18:07:19.130'
+       */
+      override def sqlType : Int = java.sql.Types.VARCHAR
+      override def setValue(v: LocalDateTime, p: PreparedStatement, idx: Int) : Unit = {
+        p.setString(idx, v.toString)
+      }
+      override def getValue(r: ResultSet, idx: Int) : LocalDateTime = {
+        LocalDateTime.parse(r.getString(idx))
+      }
+      override def updateValue(v: LocalDateTime, r: ResultSet, idx: Int) : Unit = {
+        r.updateString(idx, v.toString)
+      }
+      override def valueToSQLLiteral(value: LocalDateTime) = {
+        value match {
+          case null => "NULL"
+          case _ =>  "'" + value.toString + "'"
+        }
+      }
+    }
+
+    class OffsetTimeJdbcType extends DriverJdbcType[OffsetTime] {
+      override def sqlType : Int = {
+        /**
+         * Stores the [[OffsetTime]] as a 'VARCHAR' on databases, like SQLite, with no
+         * specific 'TIME' implementations. An example persisted value would be '18:22:43.417+01:00'.
+         */
+        java.sql.Types.VARCHAR
+      }
+      override def setValue(v: OffsetTime, p: PreparedStatement, idx: Int) : Unit = {
+        p.setString(idx, v.toString)
+      }
+      override def getValue(r: ResultSet, idx: Int) : OffsetTime = {
+        OffsetTime.parse(r.getString(idx))
+      }
+      override def updateValue(v: OffsetTime, r: ResultSet, idx: Int) : Unit = {
+        r.updateString(idx, v.toString)
+      }
+      override def valueToSQLLiteral(value: OffsetTime) = {
+        value match {
+          case null => "NULL"
+          case _ =>  "'" + value.toString + "'"
+        }
+      }
+    }
+
+    class InstantJdbcType extends DriverJdbcType[Instant] {
+      override def sqlType : Int = {
+        /**
+         * Stores the [[Instant]] as a 'VARCHAR' on databases, like SQLite, with no
+         * complex 'TIME' implementations. An example persisted value would be
+         * '2015-10-29T17:38:52.868Z'.
+         */
+        java.sql.Types.VARCHAR
+      }
+      override def setValue(v: Instant, p: PreparedStatement, idx: Int) : Unit = {
+        p.setString(idx, v.toString)
+      }
+      override def getValue(r: ResultSet, idx: Int) : Instant = {
+        Instant.parse(r.getString(idx))
+      }
+      override def updateValue(v: Instant, r: ResultSet, idx: Int) : Unit = {
+        r.updateString(idx, v.toString)
+      }
+      override def valueToSQLLiteral(value: Instant) = {
+        value match {
+          case null => "NULL"
+          case _ =>  "'" + value.toString + "'"
+        }
+      }
+    }
+
```

Comment:
there is a lot of copy and paste between the classes above, let's pull that together into a super class and not repeat